index.js

```diff
@@ -369,6 +369,34 @@
         }
       });
 
+  var reduce =
+  def('reduce',
+      {b: [Foldable]},
+      [$.Function, a, b, a],
+      function(f, initial, foldable) {
+        if (_type(foldable) === 'Array') {
+          var acc = initial;
+          for (var idx = 0; idx < foldable.length; idx += 1) {
+            acc = f(acc, foldable[idx]);
+          }
+          return acc;
+          } else {
+          return foldable.reduce(f, initial);
+        }
+      });
+
+  var add =
+  def('add',
+      {},
+      [$.FiniteNumber, $.FiniteNumber, $.FiniteNumber],
+      function(a, b) { return a + b; });
+
+  var sum =
+  def('sum',
+      {},
+      [$.Array($.FiniteNumber), $.FiniteNumber],
+      reduce(add, 0));
```

Comment:
I'd be totally okay with replacing `add` with `function(a, b) { return a + b; }` here. I'm not sure it's worth reusing a trivial function if it forces us to separate a function's implementation from its documentation. What do you think?